website/blog/deployers.md

```diff
@@ -0,0 +1,610 @@
+}
+
+// weavelet contains information about a weavelet.
+type weavelet struct {
+    envelope *envelope.Envelope // envelope to the weavelet
+    address  string             // weavelet's address
+}
+
+// handler handles messages from a weavelet.
+type handler struct {
+    deployer *deployer          // underlying deployer
+    envelope *envelope.Envelope // envelope to the weavelet
+}
+
+// Check that handler implements the envelope.EnvelopeHandler interface.
+var _ envelope.EnvelopeHandler = &handler{}
+```
+
+Next, we implement a `spawn` method that spawns a weavelet to host a component.
+
+1. To spawn the weavelet, we create an `Envelope` by calling the
+   `envelope.NewEnvelope` function. This function takes in an `EnvelopeInfo`
+   that's passed to the weavelet and an `AppConfig` that describes the
+   application.
+2. We call the `UpdateComponents` method to tell the weavelet which component to
+   run. A deployer should call `UpdateComponents` whenever there is a change to
+   the set of components a weavelet should be running.
+3. We call `envelope.Serve` to handle requests from the weavelet.
+
+```go
+// The unique id of the application deployment.
+var deploymentId = uuid.New().String()
+
+// spawn spawns the provided component (if it hasn't already spawned) and
+// returns the spawned weavelet hosting the component.
+func (d *deployer) spawn(component string) (*weavelet, error) {
+    d.mu.Lock()
+    defer d.mu.Unlock()
+
+    // Check if the component has already been spawned.
+    w, ok := d.weavelets[component]
+    if ok {
+        // The component has already been spawned.
+        return w, nil
+    }
+
+    // Spawn a weavelet to host the component.
+    info := &protos.EnvelopeInfo{
+        App:           "app",               // the application name
+        DeploymentId:  deploymentId,        // the deployment id
+        Id:            uuid.New().String(), // the weavelet id
+        SingleProcess: false,               // is the app a single process?
+        SingleMachine: true,                // is the app on a single machine?
+        RunMain:       component == "main", // should the weavelet run main?
+    }
+    config := &protos.AppConfig{
+        Name:   "app",       // the application name
+        Binary: flag.Arg(0), // the application binary
+    }
+    envelope, err := envelope.NewEnvelope(context.Background(), info, config)
+    if err != nil {
+        return nil, err
+    }
+
+    // Inform the weavelet of the component it should host.
+    if err := envelope.UpdateComponents([]string{component}); err != nil {
+        return nil, err
+    }
+
+    // Handle messages from the weavelet.
+    go envelope.Serve(&handler{d, envelope})
+
+    // Return the weavelet.
+    w = &weavelet{envelope: envelope, address: envelope.WeaveletInfo().DialAddr}
+    d.weavelets[component] = w
+    return w, nil
+}
+```
+
+Now, we implement the `EnvelopeHandler` methods.
+
+<div class="note">
```

Comment:
Is there note support built in to Github markdown? Maybe the following:

https://github.com/orgs/community/discussions/16925